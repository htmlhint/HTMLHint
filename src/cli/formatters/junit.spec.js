const ChildProcess = require('child_process')
const path = require('path')
const serializer = require('jest-serializer-path')

expect.addSnapshotSerializer(serializer)

describe('CLI', () => {
  describe('Formatter: junit', () => {
    it('should have stdout output with formatter junit', (done) => {
      ChildProcess.exec(
        [
          'node',
          path.resolve(__dirname, '../../../bin/htmlhint'),
          path.resolve(__dirname, '__fixtures__', 'example.html'),
          '--format',
          'junit',
        ].join(' '),
        (error, stdout, stderr) => {
          expect(typeof error).toBe('object')
          expect(error.code).toBe(1)

          expect(
            stdout.replace(/time="\d+\.\d+"/g, 'time="0.999"')
          ).toMatchSnapshot()

          expect(stderr).toBe('')
          done()
        }
      )
    })
  })
})
