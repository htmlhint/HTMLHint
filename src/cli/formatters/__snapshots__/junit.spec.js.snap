// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Formatter: junit should have stdout output with formatter junit 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
    <testsuite name=\\"HTMLHint Tests\\" time=\\"0.999\\" tests=\\"1\\" failures=\\"1\\">
        <testcase name=\\"<PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html\\" time=\\"0.999\\">
            <failure message=\\"Found 20 errors\\"><![CDATA[L8 |  <div bad='' bad=''></div>
          ^ The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
L8 |  <div bad='' bad=''></div>
                 ^ The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
L8 |  <div bad='' bad=''></div>
                 ^ Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
L9 |  <div bad='' bad=''></div>
          ^ The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
L9 |  <div bad='' bad=''></div>
                 ^ The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
L9 |  <div bad='' bad=''></div>
                 ^ Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
L10 |  <input type=\\"text\\"></input>
                          ^ Tag must be paired, no start tag: [ </input> ] (tag-pair)
L11 |  <div class\\"foo\\">
       ^ Special characters must be escaped : [ < ]. (spec-char-escape)
L11 |  <div class\\"foo\\">
                      ^ Special characters must be escaped : [ > ]. (spec-char-escape)
L13 |          </div>
               ^ Tag must be paired, no start tag: [ </div> ] (tag-pair)
L14 |        </div>
             ^ Tag must be paired, no start tag: [ </div> ] (tag-pair)
L15 |      </hello>
           ^ Tag must be paired, no start tag: [ </hello> ] (tag-pair)
L16 |    </test>
         ^ Tag must be paired, no start tag: [ </test> ] (tag-pair)
L17 |  </div>
       ^ Tag must be paired, no start tag: [ </div> ] (tag-pair)
L21 |          <div class='' what='' something=''></div>
                   ^ The value of attribute [ class ] must be in double quotes. (attr-value-double-quotes)
L21 |          <div class='' what='' something=''></div>
                            ^ The value of attribute [ what ] must be in double quotes. (attr-value-double-quotes)
L21 |          <div class='' what='' something=''></div>
                                    ^ The value of attribute [ something ] must be in double quotes. (attr-value-double-quotes)
L25 |  </div>
       ^ Tag must be paired, no start tag: [ </div> ] (tag-pair)
L26 |</bad>
     ^ Tag must be paired, no start tag: [ </bad> ] (tag-pair)
L27 |</bad>
     ^ Tag must be paired, no start tag: [ </bad> ] (tag-pair)]]></failure>
        </testcase>
    </testsuite>
</testsuites>
"
`;
